client/allocrunner/task_hook_coordinator.go

```diff
@@ -7,33 +7,46 @@ import (
 	"github.com/hashicorp/nomad/nomad/structs"
 )
 
-// TaskHookCoordinator helps coordinate when main start tasks can launch
+// TaskHookCoordinator helps coordinate when mainTasks start tasks can launch
 // namely after all Prestart Tasks have run, and after all BlockUntilCompleted have completed
 type taskHookCoordinator struct {
 	logger hclog.Logger
 
+	// constant for quickly starting all prestart tasks
 	closedCh chan struct{}
 
-	mainTaskCtx       context.Context
-	mainTaskCtxCancel func()
-
-	prestartSidecar   map[string]struct{}
-	prestartEphemeral map[string]struct{}
+	// when the cancel method is called, it cancels the task context, which closes the corresponding channel,
+	// which is being watched in in the task runner (in the Run() method before MAIN: in the file task_runner.go)
+	// which then unblocks that switch statement and allows the Run() code for that task to progress
```

Comment:
This seems a bit verbose or low-level and will be suseptible to code drift, and not sure if it helps me understand the code more.  Maybe something a bit more high level, like:

> Each context is used to gate task runners launching the tasks. A task runner waits until the context associated its lifecycle context is done/cancelled.

Reply:
Resolved in 368f894